environ/funcs.go

```diff
@@ -173,6 +160,27 @@ func lookUpPlugin(dirs []string, name string) (p string, err error) {
 	return
 }
 
+func convert(v interface{}) interface{} {
+	if m, ok := v.(map[string]interface{}); ok {
+		for k, v := range m {
+			m[k] = convert(v)
+		}
+	}
+	list, ok := v.([]interface{})
+	if !ok {
+		return v
+	}
+	var str []string
+	for _, val := range list {
+		s, ok := val.(string)
+		if !ok {
+			return v
```

Comment:
Again my fault... it just occurred to me that we're not handling lists of json objects, so what we need to do is, if they're all strings, return a slice of strings.  If they're not strings, return a slice of interfaces, where each interface has been converted.